src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -74,136 +74,48 @@ QtObject:
   QtProperty[QVariant] recipientsModel:
     read = getRecipientsModel
 
-  proc buildMultiTransactionExtraData(self: Controller, metadata: backend_activity.ActivityEntry, item: MultiTransactionDto): ExtraData =
+  proc buildMultiTransactionExtraData(self: Controller, metadata: backend_activity.ActivityEntry): ExtraData =
     if metadata.symbolIn.isSome():
       result.inAmount = self.currencyService.parseCurrencyValue(metadata.symbolIn.get(), metadata.amountIn)
     if metadata.symbolOut.isSome():
       result.outAmount = self.currencyService.parseCurrencyValue(metadata.symbolOut.get(), metadata.amountOut)
 
-  proc buildTransactionExtraData(self: Controller, metadata: backend_activity.ActivityEntry, item: ref TransactionDto): ExtraData =
+  proc buildTransactionExtraData(self: Controller, metadata: backend_activity.ActivityEntry): ExtraData =
     if metadata.symbolIn.isSome():
       result.inAmount = self.currencyService.parseCurrencyValue(metadata.symbolIn.get(), metadata.amountIn)
     if metadata.symbolOut.isSome():
       result.outAmount = self.currencyService.parseCurrencyValue(metadata.symbolOut.get(), metadata.amountOut)
 
-  proc getResolvedSymbol(self: Controller, transaction: TransactionDto): string =
-    if transaction.symbol != "":
-      result = transaction.symbol
-    else:
-      let contractSymbol = self.tokenService.findTokenSymbolByAddress(transaction.contract)
-      if contractSymbol != "":
-        result = contractSymbol
-      else:
-        result = "ETH"
-
   proc backendToPresentation(self: Controller, backendEntities: seq[backend_activity.ActivityEntry]): seq[entry.ActivityEntry] =
-    var multiTransactionsIds: seq[int] = @[]
-    var transactionIdentities: seq[backend.TransactionIdentity] = @[]
-    var pendingTransactionIdentities: seq[backend.TransactionIdentity] = @[]
-
-    # Extract metadata required to fetch details
-    # TODO: see #11598. Temporary here to show the working API. Details for each entry will be done as required
-    # on a detail request from UI after metadata is extended to include the required info
+    let amountToCurrencyConvertor = proc(amount: UInt256, symbol: string): CurrencyAmount =
+      return currencyAmountToItem(self.currencyService.parseCurrencyValue(symbol, amount),
+                                self.currencyService.getCurrencyFormat(symbol))
     for backendEntry in backendEntities:
-      case backendEntry.payloadType:
+      var ae: entry.ActivityEntry
+      case backendEntry.getPayloadType():
         of MultiTransaction:
-          multiTransactionsIds.add(backendEntry.id)
+          let extraData = self.buildMultiTransactionExtraData(backendEntry)
+          ae = entry.newMultiTransactionActivityEntry(backendEntry, extraData, amountToCurrencyConvertor)
         of SimpleTransaction:
```

Comment:
nit: this case and the one below look identical, could comma-separate the two enum values